react/src/Header.js

```diff
@@ -0,0 +1,258 @@
+import './css/App.css';
+import React from 'react';
+import { useContext } from 'react';
+import { Link, NavLink, useNavigate } from 'react-router-dom';
+import HeaderData from './HeaderData';
+import { BarSpinner, StandardSpinner } from './Spinners';
+import Auth from './utils/Auth';
+import Client from './utils/Client';
+import Context from './utils/Context';
+import LiveTime from './LiveTime';
+import Env from './utils/Env';
+import Image from './utils/Image';
+import Logout from './utils/Logout';
+import { ReadOnlyModeDisplay } from './ReadOnlyMode';
+import { useFetching } from './utils/Fetch';
+// import JustLoggedIn from './JustLoggedIn';
+// Issues with serving images ONLY from 4dn-dcic/dev NOT from cgap-supertest ...
+// So serve from my GitHub account for now ...
+// import LockImage from './media/lock.jpg';
+// import UnlockImage from './media/unlock.jpg';
+
+const Header = (props) => {
+
+    const [ header ] = useContext(HeaderData);
```

Comment:
I am using the useContext mechanism only for the global "header" data/state, which is shared across _all_ components; so since this one is shared by all, I don't mind having to wrap the App in the "Provider" element for this global context HeaderData.